File: 2022/day1/part1.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>

int greaterUntilNow = 0;

int sumItems(std::vector<int> &array) {
	int sum = 0;
	for (unsigned int i = 0; i < array.size(); i++) {
		sum += array[i];
	}
	
	return sum;
}

int main() {
	std::string b;
	std::ifstream myFile("inputs.txt");
	std::vector<int> arr = {};
	
	while (getline(myFile, b)) {
		if (b == "") {
			int theVeryCoolSum = sumItems(arr);
			if (theVeryCoolSum > greaterUntilNow) {
				greaterUntilNow = theVeryCoolSum;
				// std::cout <<  << std::endl;
			}

			arr.clear();
			continue;
		} else {
			arr.push_back(std::stoi(b));
		}
	}
	
	myFile.close();

	std::cout << greaterUntilNow << std::endl;

	return 0;
}

File: 2022/day1/part1.py
greater_until_now = 0

def sum_items(arr):
	sum = 0
	for i in arr:
		sum += i
	return sum

with open('inputs.txt', 'r') as f:
	b = f.read().splitlines()

	arr = []
	for i in b:
		if (i == ''):
			the_very_cool_sum = sum_items(arr)
			if (the_very_cool_sum > greater_until_now):
				greater_until_now = the_very_cool_sum

			arr = [] # preparando o array para uma nova checagem
			continue
		else:
			arr.append(int(i))

print(greater_until_now) # the answer omg